### PR TITLE
Implement post creation

### PR DESCRIPTION
**Pull request by writeriks**
Changes to create post

https://github.com/user-attachments/assets/81bcc18f-9428-4f3a-a05b-fab9b1cddb86

### CODE REVIEW

File: src/components/new-post/new-post.tsx [hunk 1]
```diff
@@ -10,27 +10,88 @@ import { Button } from '@/components/ui/button';
...
 
   const fileInputRef = useRef<HTMLInputElement>(null);
 
+  const mutation = useMutation({
+    mutationFn: ({ post, postImageData }: { post: Post; postImageData: string }) =>
+      postService.createNewPost(post, postImageData),
+    onSuccess: data => {
+      dispatch(
+        setUINotification({
+          message: 'Post başarıyla oluşturuldu.',
```
Comment (sakirdemirhan) [inline src/components/new-post/new-post.tsx:40]:
```suggestion
          message: 'Gönderi başarıyla oluşturuldu.',
```

File: src/components/new-post/new-post.tsx [hunk 2]
```diff
@@ -10,27 +10,88 @@ import { Button } from '@/components/ui/button';
...
+          notificationType: UINotificationEnum.SUCCESS,
+        })
+      );
+      console.log('Post created successfully:', data);
```
Comment (sakirdemirhan) [inline src/components/new-post/new-post.tsx:44]: Leftover console.log

File: src/components/new-post/new-post.tsx [hunk 3]
```diff
@@ -84,19 +150,26 @@ const NewPost = (): React.ReactElement => {
...
             </Button>
 
             <Button
               className='flex items-center min-w-24 px-4 text-lg py-2 rounded-full'
               variant='default'
+              onClick={submitPost}
+              disabled={isSubmitDisabled}
             >
-              Gönder
+              {mutation.isPending ? 'Gönderiliyor' : 'Gönder'}
```
Comment (sakirdemirhan) [inline src/components/new-post/new-post.tsx:172]: optional suggestion
```suggestion
              {mutation.isPending ?  <Icons.spinner className='mr-2 h-4 w-4 animate-spin' /> 'Gönderiliyor' : 'Gönder'}
```

File: src/components/new-post/new-post.tsx [hunk 4]
```diff
@@ -10,27 +10,88 @@ import { Button } from '@/components/ui/button';
...
+        setCashTags(filteredCashTags);
+      }
+    });
+  }, [cashTags, content]);
+
+  const submitPost = async (): Promise<void> => {
+    const post: Post = {
+      userId: currentUser.userId,
+      stockTickers: cashTags,
+      createdAt: Timestamp.fromMillis(Date.now()),
```
Comment (sakirdemirhan) [inline src/components/new-post/new-post.tsx:87]: Would `Timestamp.now()` work?

File: src/services/firebase-service/types/db-types/post.ts [hunk 5]
```diff
@@ -8,20 +8,21 @@ export type PostCollection = Post[];
 export type PostId = string;
 
 export interface Post {
-  postId: PostId;
+  postId?: PostId;
```
Comment (sakirdemirhan) [inline src/services/firebase-service/types/db-types/post.ts:11]: I think `postId` should not be optional.

File: src/services/firebase-service/firebase-operations.ts [hunk 6]
```diff
@@ -129,6 +131,19 @@ class FirebaseOperations {
...
   };
 
+  /*
+   * Uploads an image to Firebase Storage from client side
+   * Will not work on server side call due to firebase storage rules
+   * @param media - The media data
+   * @param imageData - The image data
+   * @returns The download URL of the image
+   */
+  handleSubmitImageFromClient = async (media: MediaData, imageData: string): Promise<string> => {
```
Comment (sakirdemirhan) [inline src/services/firebase-service/firebase-operations.ts:141]: Is this still needed?

File: src/components/new-post/new-post.tsx (hunk 4 above)
Comment (writeriks) [reply src/components/new-post/new-post.tsx:87]: seems to work

File: src/services/firebase-service/firebase-operations.ts (hunk 6 above)
Comment (writeriks) [reply src/services/firebase-service/firebase-operations.ts:141]: not at the moment, removing it.

File: src/services/firebase-service/types/db-types/post.ts (hunk 5 above)
Comment (writeriks) [reply src/services/firebase-service/types/db-types/post.ts:11]: when you create a new post, currently, it creates with an autogenerated id. So when you create a post object, there is no id in it. Im not sure how to change it

**Pull request merged**